extensions/amp-story/1.0/amp-story.js

```diff
@@ -2471,28 +2446,99 @@ export class AmpStory extends AMP.BaseElement {
+        .registerLocalizedStringBundle('en-GB', LocalizedStringsEnGb)
+        .registerLocalizedStringBundle('es', LocalizedStringsEs)
+        .registerLocalizedStringBundle('es-419', LocalizedStringsEs419)
+        .registerLocalizedStringBundle('fr', LocalizedStringsFr)
+        .registerLocalizedStringBundle('hi', LocalizedStringsHi)
+        .registerLocalizedStringBundle('id', LocalizedStringsId)
+        .registerLocalizedStringBundle('it', LocalizedStringsIt)
+        .registerLocalizedStringBundle('ja', LocalizedStringsJa)
+        .registerLocalizedStringBundle('ko', LocalizedStringsKo)
+        .registerLocalizedStringBundle('nl', LocalizedStringsNl)
+        .registerLocalizedStringBundle('no', LocalizedStringsNo)
+        .registerLocalizedStringBundle('pt-PT', LocalizedStringsPtPt)
+        .registerLocalizedStringBundle('pt-BR', LocalizedStringsPtBr)
+        .registerLocalizedStringBundle('ru', LocalizedStringsRu)
+        .registerLocalizedStringBundle('tr', LocalizedStringsTr)
+        .registerLocalizedStringBundle('vi', LocalizedStringsVi)
+        .registerLocalizedStringBundle('zh-CN', LocalizedStringsZhCn)
+        .registerLocalizedStringBundle('zh-TW', LocalizedStringsZhTw)
+        .registerLocalizedStringBundle(
+          'en-xa',
+          createPseudoLocale(LocalizedStringsEn, (s) => `[${s} one two]`)
+        );
+    }
+  }
+
   /**
    * If there are inline localization strings, register as current document language.
+   * @param {string} languageCode
+   * @return {boolean}
    */
-  registerInlineLocalizationStrings_() {
+  registerInlineLocalizationStrings_(languageCode) {
     const inlineStringsEl = this.win.document.querySelector(
       'script[amp-localization="amp-story"]'
     );
     if (
       inlineStringsEl?.getAttribute('i-amphtml-version') !==
       getMode(this.win).rtvVersion
     ) {
-      return;
+      return false;
     }
     const stringsOrNull = tryParseJson(inlineStringsEl.textContent);
+
     if (!stringsOrNull) {
-      return;
+      return false;
     }
     const localizationService = getLocalizationService(this.element);
     localizationService.registerLocalizedStringBundle(
-      localizationService.getLanguageCodesForElement(this.element)[0],
+      languageCode,
       stringsOrNull
     );
+    return true;
+  }
+
+  /**
+   * Fetches from the CDN or localhost the localization strings.
+   * @param {string} languageCode
+   */
+  fetchLocalizationStrings_(languageCode) {
+    const localizationService = getLocalizationService(this.element);
+
+    const localizationUrl = `${calculateScriptBaseUrl(
+      this.win.location,
+      getMode(this.win).localDev
+    )}/v0/amp-story.${languageCode}.json`;
+
+    Services.xhrFor(this.win)
+      .fetchJson(localizationUrl)
+      .then((res) => res.json())
+      .then((json) =>
+        localizationService.registerLocalizedStringBundle(languageCode, json)
+      )
+      .catch((err) => {
```

Comment:
When does this happen, if the AMP Cache falls back to english for an unknown `languageCode`?

Reply:
Locally you could create a demo with a language that doesn't exist and this will show in the console. Also the cache doesn't do this now, but will be implemented soon. Also if the language doesn't have a valid format (let's say someone uses `../../..` or `en/us` or some other complex string that the cache might not be able to fallback due to the path being broken).

All these can avoid breaking the code by catching the error. We could use regex for some of these, but in case anything goes wrong, catching the error makes the most sense IMO.

Reply:
We shouldn't launch this till the AMP Cache does this fallback then, otherwise all docs with wrong `lang` parameters would start failing.

Reply:
Yeah we can trigger the experiment flag when the cache works better. This `catch` is still useful if the language breaks the url, the network drops, etc. Having a regex (`\w+(-\w+)*`) would fix the former but not the latter.